core/shared/src/main/scala/fs2/async/immutable/Signal.scala

```diff
@@ -37,6 +38,49 @@ abstract class Signal[F[_], A] { self =>
 }
 
 object Signal {
+  implicit def signalIsFunctor[F[_]: Functor]: Functor[({ type L[X] = Signal[F, X] })#L] =
+    new Functor[({ type L[X] = Signal[F, X] })#L] {
+      override def map[A, B](fa: Signal[F, A])(f: A => B): Signal[F, B] = fa.map(f)
+    }
+
+  implicit def signalIsApplicative[F[_]](
+      implicit effectEv: Effect[F],
```

Comment:
Note on series/1.0, we'll likely be able to get rid of the EC requirement and downgrade `Effect` to `Concurrent`